File: src/Contexts/Rutine/Exercises/domain/ExerciseMuscle.ts
import { InvalidArgumentError } from '../../../Shared/domain/value-object/InvalidArgumentError';
import { StringValueObject } from '../../../Shared/domain/value-object/StringValueObject';

export class ExerciseMuscle extends StringValueObject {
  constructor(value: string) {
    super(value);
    this.ensureLengthIsLessThan30Characters(value);
  }

  private ensureLengthIsLessThan30Characters(value: string): void {
    if (value.length > 30) {
      throw new InvalidArgumentError(`The Exercise Name <${value}> has more than 30 characters`);
    }
  }
}

File: src/main/java/com/kuro/rutineweb/exercises/entities/ExerciseMuscle.java
package com.kuro.rutineweb.exercises.entities;

import com.kuro.rutineweb.Shared.entities.valueObject.InvalidArgumentError;
import com.kuro.rutineweb.Shared.entities.valueObject.StringValueObject;

public class ExerciseMuscle extends StringValueObject {
    public ExerciseMuscle(String value) {
        super(value);
        ensureLengthIsLessThan30Characters(value);
    }

    private void ensureLengthIsLessThan30Characters(String value) {
        if(value.length() > 30) {
            throw new InvalidArgumentError("The Exercise Muscle " + value + " has more than 30 characters");
        }
    }
}
